privateSBOMExchange/src/petra/lib/models/__init__.py

```diff
@@ -61,3 +65,166 @@ def tree_from_nodes(nodes, values, root):
+    def visit_root(self, root):
+        # Compute hash for the root using its data and the hashes of its children
+        children_hashes = b''.join(child.accept(self) for child in root.children)
+        combined_data = root.data.encode() + children_hashes
+        return hashlib.sha256(cpabe(combined_data)).digest()
+class PrintVisitor:
+    """Visitor that prints the data of each node."""
+    def visit_leaf(self, leaf):
+        print(f"Leaf: {leaf.data}")
+
+    def visit_internal_node(self, node):
+        print(f"Internal Node: {node.data}")  # Print package data
+        for child in node.children:
+            child.accept(self)
+    def visit_root(self, root):
+        print(f"Root: {root.data}")  # Print root data
+        for child in root.children:
+            child.accept(self)        
+def build_sbom_tree(doc):
+    """Builds a Merkle tree from an SPDX SBOM."""
+    
+    leaves = []
+    #create internal nodes for each field
+    document_info=doc.get_document()
+    for field_name, field_value in document_info.items():
+#        print(f"Field{field_name}{field_value}")
+        if not field_name.startswith('_'):
+            leaves.append(Leaf(f"Field{field_name}{field_value}"))
+
+    # Create internal nodes for each package
+    pkgs=doc.get_packages()
+    root_children = []
+
+    for package in pkgs:
+        pkg_nodes=[]
+        # Extract fields for each package
+        package_fields = [
+            
+            Leaf(f"Field{key}{value}")
+            for key, value in package.items()
+            if not key.startswith('_')  # Exclude internal attributes
+        ]
+        root_children.append(InternalNode(package["name"], package_fields))
+
+
+    files=doc.get_files()
+    if (files):
+        for file in files:
+            leaves.append(Leaf(f"Fieldfile{file}"))
+        
+
+    licenses=doc.get_licenses()
+    if (licenses):
+        for license in licenses:
+            leaves.append(Leaf(f"Fieldlicense{license}"))
+        
+    vulnarabilities=doc.get_vulnerabilities()
+    if (vulnarabilities):
+        for vulnarability in vulnarabilities:
+            leaves.append(Leaf(f"Fieldvulnarability{vulnarability}"))
+          
+            
+    relationships=doc.get_relationships()
+    if (relationships):
+        for relationship in relationships:
+            leaves.append(Leaf(f"Fieldrelationship{relationship}"))
+          
+                
+    services=doc.get_services()
+    if(services):
+        for service in services:
+            leaves.append(Leaf(f"Fieldservice{service}"))
+          
+             
+    # Create the root node containing all leaves and package nodes
+    sbom_name=doc.get_document()["name"]
+    root = RootNode(sbom_name,leaves + root_children)
+    return root
+
+
+
+# Parse SPDX data into a Document object
+
+SBOM_parser = SBOMParser()
+#SBOM_parser.parse_file("simple_sbom.json")
+   
+SBOM_parser.parse_file("../sbom_data/bom-shelter/in-the-wild/spdx/julia.spdx.json")   
```

Comment:
This shouldn't be part of the init. Make it a test instead.

Reply:
@SantiagoTorres  Thank you for the feedback! I've pushed the changes you suggested.